frontend/components/CameraView.js

```diff
@@ -0,0 +1,98 @@
+import { Camera, Permissions } from 'expo';
+import { Text, Button } from 'react-native-elements';
+import { Ionicons } from '@expo/vector-icons';
+
+export default class CameraView extends React.Component {
+
+  state = {
+    hasCameraPermission: null,
+    type: Camera.Constants.Type.back,
+  };
+
+  async componentDidMount() {
+    const { status } = await Permissions.askAsync(Permissions.CAMERA);
+    this.setState({ hasCameraPermission: status === 'granted' });
+  }
+
+  handlePictureTaken = (image) => {
+    // When the image data is returned to us then exit the camera view and pass the image back to the main view
+    this.props.navigation.goBack();
+    this.props.navigation.getParam('handlePictureTaken', () => {alert("Reached camera from unknown route")})(image);
+  }
+
+  render() {
+    const { hasCameraPermission } = this.state;
+    // Check that permission was granted
+    if (hasCameraPermission === null) {
+      return <View />;
+    } else if (hasCameraPermission === false) {
+      return <Text>No access to camera</Text>;
+    } else {
+      return (
+        <View style={styles.container}>
+          <Camera style={styles.container} 
+            type={this.state.type}
+            ref={ref => {this.camera = ref;}}>
+              {this.state.loading ? (
+                <View style={styles.imageOverlay}>
+                    <ActivityIndicator size={"large"}/>
+                </View>
+              ) : (
+                <View style={styles.imageOverlay}>
+                  <TouchableOpacity 
+                    style={[styles.overlayButton, styles.alignCenter]}
+                    onPress={() => {
+                      this.setState({
+                          type: this.state.type === Camera.Constants.Type.back
+                          ? Camera.Constants.Type.front
+                          : Camera.Constants.Type.back,
+                      });
+                    }}>
+                      <Text style={{ fontSize: 18, marginBottom: 10, color: 'white' }}>
+                        {' '}Flip{' '}
+                      </Text>
+                  </TouchableOpacity>
+                  <Button
+                      title="  Take Picture"
+                      raised
+                      icon={<Ionicons name="ios-camera" size={24} color={"white"}/>}
+                      containerStyle={styles.overlayButton}
+                      onPress={() => {
+                          this.setState({loading: true});
+                          this.camera.takePictureAsync({
+                              quality: 0.1,
+                              base64: true,
+                              onPictureSaved: this.handlePictureTaken
+                          });
+                      }}/>
+                  <View style={styles.overlayButton}></View>
+                </View>
+              )}
+          </Camera>
+        </View>
+      );
+    }
+  }
+}
+
+
+const styles = StyleSheet.create({
```

Comment:
In order to keep your files short and easy to read, you could create a separate file for your styles.

Reply:
I agree it could shorten the length of the files but I think for our team it works better to have styles and content in the same file as they are very closely related and the styles are usually not too long.